### PR TITLE
webex: init at 42.8.0.22907

### PR DESCRIPTION
**Pull request by uvNikita**
###### Description of changes

Picking up efforts from #129851.

Seems like there are three channels:

- gold: 42.6.0.22565
- green: 42.7.0.22904
- blue: 42.8.0.22907

I selected the **blue** channel for now (chat rendering in wayland was fixed in this version for me), but let me know if you think we should use another one.

`update.sh` script is used to get newest package version from the channel. Please let me know if there is a way to integrate it with @r-ryantm so that we don't have to run the script manually. @ryantm

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/webex/version.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "version": "42.8.0.22907",
+  "packageLocation": "https://binaries.webex.com/WebexDesktop-Ubuntu-Blue/20220712081040/Webex_ubuntu.7z",
+  "checksum": "b83950cdcf978a3beda93de27b25b70554fc82fcf072a5a7ea858d2ce0d176ac"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/webex/version.json:4]: I don't think we need this file. We can just update the information in the nix file.

File: pkgs/applications/networking/instant-messengers/webex/default.nix [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    cp -r opt "$out"
+
+    ln -s "$out/opt/Webex/bin/CiscoCollabHost" "$out/bin/webex"
+    chmod +x $out/bin/webex
+
+    ln -s "$out/opt/Webex/bin/webex.desktop" "$out/share/applications/webex.desktop"
+  '';
+
+  meta = with lib; {
+    description = "Webex";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/webex/default.nix:161]: Just the program name is to short for a description.

File: pkgs/applications/networking/instant-messengers/webex/default.nix [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      opt/Webex/bin/CiscoCollabHostCefWM \
+      opt/Webex/bin/WebexFileSelector \
+      opt/Webex/bin/pxgsettings
+    for each in $(find opt/Webex -type f | grep \\.so); do
+      patchelf --set-rpath "${libPath}" "$each"
+    done
+  '';
+
+  installPhase = ''
+    mkdir -p "$out" "$out/bin" "$out/share/applications"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/webex/default.nix:151]:
```suggestion
    mkdir -p "$out/bin" "$out/share/applications"
```

File: pkgs/applications/networking/instant-messengers/webex/default.nix [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    cp -r opt "$out"
+
+    ln -s "$out/opt/Webex/bin/CiscoCollabHost" "$out/bin/webex"
+    chmod +x $out/bin/webex
+
+    ln -s "$out/opt/Webex/bin/webex.desktop" "$out/share/applications/webex.desktop"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/webex/default.nix:157]:
```suggestion
    mv "$out/opt/Webex/bin/webex.desktop" "$out/share/applications/webex.desktop"
```
or does that not work?

File: pkgs/applications/networking/instant-messengers/webex/default.nix [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    xcbutilkeysyms
+    xcbutilrenderutil
+    xcbutilwm
+    p7zip
+    wayland
+  ];
+
+  libPath = "$out/opt/Webex/lib:"
+    + "$out/opt/Webex/bin:"
+    + lib.makeLibraryPath buildInputs;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/webex/default.nix:123]:
```suggestion
  libPath = "$out/opt/Webex/lib:$out/opt/Webex/bin:{lib.makeLibraryPath buildInputs}";
```

File: pkgs/applications/networking/instant-messengers/webex/default.nix [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+  pname = "Webex";
+  version = versionSpec.version;
+
+  src = fetchurl {
+    url = versionSpec.packageLocation;
+    sha256 = versionSpec.checksum;
+  };
+
+  nativeBuildInputs = [ patchelf ];
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/webex/default.nix:69]:
```suggestion
```

part of stdenv

File: pkgs/applications/networking/instant-messengers/webex/default.nix [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+, wayland
+, libXScrnSaver
+}:
+
+let
+  inherit (lib) optional;
+  versionSpec = builtins.fromJSON (builtins.readFile ./version.json);
+in
+stdenv.mkDerivation rec {
+  pname = "Webex";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/webex/default.nix:60]:
```suggestion
  pname = "webex";
```

File: pkgs/applications/networking/instant-messengers/webex/default.nix [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+, xcbutilrenderutil
+, xcbutilwm
+, p7zip
+, wayland
+, libXScrnSaver
+}:
+
+let
+  inherit (lib) optional;
+  versionSpec = builtins.fromJSON (builtins.readFile ./version.json);
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/webex/default.nix:57]:
```suggestion
  versionSpec = lib.importJSON ./version.json;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by uvNikita** [comment]
Thank you for such a quick review, @SuperSandro2000!

I added all your suggestions.
I just learnt about `passthru.updateScript`, so moved `update.sh` script there. Also used `update-source-version` to update version, url, and checksum in `webex.nix` file.

File: pkgs/top-level/all-packages.nix [hunk 9]
```diff
@@ -31030,6 +31030,8 @@ with pkgs;
 
   webcamoid = libsForQt5.callPackage ../applications/video/webcamoid { };
 
+  webex = callPackage ../applications/networking/instant-messengers/webex.nix {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:31033]:
```suggestion
  webex = callPackage ../applications/networking/instant-messengers/webex {};
```
place the nix file at ...../webex/default.nix

File: pkgs/applications/networking/instant-messengers/webex.nix [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+  meta = with lib; {
+    description = "The all-in-one app to call, meet, message, and get work done";
+    homepage = "https://webex.com/";
+    downloadPage = "https://www.webex.com/downloads.html";
+    license = licenses.unfree;
+    maintainers = with lib.maintainers; [ uvnikita ];
+    platforms = [ "x86_64-linux" ];
+  };
+
+  passthru.updateScript = writeScript "webex-update-script" ''
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/webex.nix:161]: Please swap meta and apssthru. meta should always be last.

File: pkgs/top-level/all-packages.nix (hunk 9 above)
Comment (uvNikita) [reply pkgs/top-level/all-packages.nix:31033]: done

File: pkgs/applications/networking/instant-messengers/webex.nix (hunk 10 above)
Comment (uvNikita) [reply pkgs/applications/networking/instant-messengers/webex.nix]: done

**Pull request merged**